File: LeetCode/Medium/1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.java
class Solution {
    public int maxOperations(int[] nums, int k) {

        if(nums.length == 0) return 0;

         Arrays.sort(nums);

        int ans = 0;
        int left = 0;
        int right = nums.length - 1;

        while(left < right) {
            
            int sum = nums[left] + nums[right];

            if(sum == k) {
                ans++;
                left++;
                right--;
            }
            else if (sum < k) left++;
            else right--;
        }

        return ans;
    }
}

File: LeetCode/Medium/1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.kt
class Solution {
    fun maxOperations(nums: IntArray, k: Int): Int {
        
        if(nums.size == 0) return 0
        
        nums.sort()
        
        var ans = 0
        var left = 0
        var right = nums.size - 1

        while(left < right) {
            
            val sum = nums[left] + nums[right]
            /*
            print(left)
            print(" / ")
            println(right)

            print(nums[left])
            print(" + ")
            print(nums[right])
            print(" = ")
            println(sum)
            */

            if(sum == k) {
                ans++
                left++
                right--
            }
            else if (sum < k) left++
            else right--
        }
        
        return ans
        
    }
}